### ISSUE TITLE
Changes in biblatex 3.3

### ISSUE THREAD

**Issue by moewew** [moewew/biblatex-trad#25]
With biblatex 3.3 the name formatting macros have been changed drastically, see https://github.com/plk/biblatex/issues/372. 

The format `abbrv` probably needs to be something like

```
\DeclareNameFormat{abbrv}{%
  \nameparts{#1}%
  \usebibmacro{name:given-family}
      {\namepartfamily}
      {\namepartgiveni}
      {\namepartprefixi}
      {\namepartsuffixi}%
  \usebibmacro{name:andothers}}
```

**Comment by russella** [comment]
Dear Moewew,
Is this still an issue with 3.4? If so, it seems important to fix it. If you could give me some instructions on fixing it, I'll give it a try.
Thanks!  Alex

**Comment by moewew** [comment]
The style has aged quite well, so the only thing that really needs changing for v. 3.3 - as far as I can see - is

```
\DeclareNameFormat{abbrv}{%
  \usebibmacro{name:first-last}{#1}{#4}{#6}{#8}%
  \usebibmacro{name:andothers}}
```
in [`trad-abbrv.bbx`](https://github.com/marcodaniel/trad-biblatex/blob/master/trad-abbrv.bbx), and the code above should be the correct code to replace this.

For version 3.4 we need to change all occurrences of `prefixnumber` to `labelprefix`, that will need changes in more files, though. The general work that needs to be done is quite straightforward.

**Comment by larseggert** [comment]
Will this get fixed? At the moment, trad-abbrv is still broken.

**Comment by russella** [comment]
I've just make a pull request that implements moewew's suggested fix. It seems to work with the original test cases. @larseggert , it would be great if you could pull down my repo and see that it works with your examples as well.

**Comment by moewew** [comment]
As you noted, a similar pull request has already been started by @gvdgdo (https://github.com/marcodaniel/trad-biblatex/pull/26). The solution there even retains backwards compatibility, so I would say it is preferable.

**Comment by russella** [comment]
Moewew,
I'm worried that Marco Daniel may not be part of the github community anymore. It seems his account has been inactive for over a year. Would it make sense for one of us to merge in the pulls above--just to bring the package up to compatibility with 3.3--and make it available as a comprehensive fork?
If you are willing to host the new repository on your github site, I am willing to merge the pull req.'s and get it ready for you.
If M. Daniel returns to lead the project, of course we can hand leadership back to him.
Best, Alex

**Comment by gvdgdo** [comment]
I have the same fear (that Marco is no longer maintaining the package). To me, it would make sense to merge the pull request,  and upload the revised package to CTAN, so it can make the cutoff for TeXLive 2016.  

Guido

**Comment by moewew** [comment]
On TeX.SX Marco Daniel was last seen in October 2015. I don't know if there has been any significant attempts to try to get into contact with him.

I think it would be problematic from a license point of view to just upload the package with a fix to CTAN (at least if the name is not changed). The LPPL allows for change of maintainers, but even if that procedure is started now and a volunteer is found to take over maintainership of the package, I don't think the update can make it to TeX live 2016 in time for the cut-off.

**Comment by josephwright** [comment]
Note that the LPPL doesn't directly impact here: CTAN have their own rules to avoid 'helpful' uploads by people other than the maintainers. The LPPL requires that notice is given before taking over as maintainer, though it is possible of course to fork at any stage. CTAN are pretty unlikely to take a fork without a rename, though this can be achieved where there is good reason (see _e.g._ `biblatex` situation).

I'd suggest that if someone is willing to 'pick up' then notice is given in the usual places (c.t.t., TeXhax, ...) and CTAN are informed. It will be a while before that can 'bite' though (the assumption has always been three months after giving notice).

**Comment by josephwright** [comment]
Checking, the 3 month requirement is in the LPPL: http://www.latex-project.org/lppl.txt, `MAINTENANCE OF THE WORK`, item 4.

**Comment by russella** [comment]
I guess the most attractive possibility would be to raise Marco himself; I'll see if I can communicate with him somehow on TeX.SX or find him some other way, perhaps just via a google search. If any of you know of other online presences of his, please do reach out (or let me know).

About TL2016, I thought that various levels of package/system freeze were long since behind us. Even if the package were to miss the deadline, I suppose it could be shipped as an update, right?

Finally, if we can't reach Marco, I do think it would be useful to have an updated repository somewhere (especially since the changes in the current pull requests are so minimal).

**Comment by moewew** [comment]
You could at try to reach him on the mail address from the documentations of his packages http://mirrors.ctan.org/macros/latex/contrib/biblatex-contrib/biblatex-trad/biblatex-trad.pdf

**Comment by russella** [comment]
Good idea.

**Comment by russella** [comment]
@moewew I've been in touch with Marco. It sounds as though time is a scarce resource right now, and that he won't be able to keep after trad-biblatex. Could we discuss things briefly? When you have a minute, please send me an email at alexandercraigrussell@gmail.com

**Comment by moewew** [comment]
Marco has handed over maintenance of the package to me. I intend to publish an updated version of the package that works with biblatex 3.3 and 3.4 as soon as possible.

**Comment by moewew** [comment]
A first attempt to get the name format on track again is in 8753f52. Testing (especially from those who are still on version 3.3 of `biblatex`) would be highly appreciated.

**Comment by moewew** [comment]
The code now uses `labelprefix` instead of `prefixnumber` with `biblatex ` >= 3.4.

Again, testing would be highly appreciated, so we can try to ship the update to CTAN in a few days.

**Comment by moewew** [comment]
This should be fixed in version 0.3 which has been on CTAN for a bit now. Testing is, of course, still very much appreciated.
